script.js

```diff
@@ -1 +1,519 @@
+
+  numOfClicks = 0;
+  score = 0;
+
+  // Refresh position of card clicked and prevent user from clicking on the same card multiple times to score
+  positionOfCardClicked.column = '';
+  positionOfCardClicked.row = '';
+};
+
+const getRandomIndex = (size) => Math.floor(Math.random() * size);
+
+const makeDeck = () => {
+  // create the empty deck at the beginning
+  const newDeck = [];
+  const suits = ['hearts', 'diamonds', 'clubs', 'spades'];
+
+  for (let suitIndex = 0; suitIndex < suits.length; suitIndex += 1) {
+    // make a variable of the current suit
+    const currentSuit = suits[suitIndex];
+
+    // loop to create all cards in this suit
+    // rank 1-13
+    for (let rankCounter = 1; rankCounter <= 13; rankCounter += 1) {
+      // Convert rankCounter to string
+      let cardName = `${rankCounter}`;
+      // declare displayName
+      let displayName;
+
+      // 1, 11, 12 ,13
+      if (cardName === '1') {
+        cardName = 'ace';
+      } else if (cardName === '11') {
+        cardName = 'jack';
+      } else if (cardName === '12') {
+        cardName = 'queen';
+      } else if (cardName === '13') {
+        cardName = 'king';
+      }
+
+      if (cardName === 'ace') {
+        displayName = 'A';
+      } else if (cardName === 'jack') {
+        displayName = 'J';
+      } else if (cardName === 'queen') {
+        displayName = 'Q';
+      } else if (cardName === 'king') {
+        displayName = 'K';
+      } else {
+        displayName = cardName;
+      }
+      let suitSymbol;
+      let colorX;
+      if (suitIndex === 0) {
+        suitSymbol = '❤️';
+        colorX = 'red';
+      } else if (suitIndex === 1) {
+        suitSymbol = '◆';
+        colorX = 'red';
+      } else if (suitIndex === 2) {
+        suitSymbol = '♣️';
+        colorX = 'black';
+      } else {
+        suitSymbol = '♠️';
+        colorX = 'black';
+      }
+      // make a single card object variable
+      const card = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+        display: displayName,
+        symbol: suitSymbol,
+        color: colorX,
+      };
+
+      // add the card to the deck
+      newDeck.push(card); // add double the cards to the deck
+      newDeck.push(card);
+    }
+  }
+
+  return newDeck;
+};
+const shuffleCards = (cards) => {
+  // loop over the entire cards array
+  for (let currentIndex = 0; currentIndex < cards.length; currentIndex += 1) {
+    // select a random position from the deck
+    const randomIndex = getRandomIndex(cards.length);
+
+    // get the current card in the loop
+    const currentItem = cards[currentIndex];
+
+    // get the random card
+    const randomItem = cards[randomIndex];
+
+    // swap the current card and the random card
+    cards[currentIndex] = randomItem;
+    cards[randomIndex] = currentItem;
+  }
+
+  // give back the shuffled deck
+  return cards;
+};
+
+// Fn to display card in squareClick method
```

Comment:
It's obvious that this is a fn. Comment can just be imperative what the fn does, e.g. "Display card in squareClick method"